mne_connectivity/base.py

```diff
@@ -0,0 +1,548 @@
+                             f'"raveled, "full", not {output}.')
+
+        if output == 'raveled':
+            data = self._data
+        else:
+            # get the new shape of the data array
+            if self.is_epoched:
+                new_shape = [self.n_epochs]
+            else:
+                new_shape = []
+            new_shape.extend([self.n_nodes, self.n_nodes])
+            if 'freqs' in self.dims:
+                new_shape.append(len(self.coords['freqs']))
+            if 'times' in self.dims:
+                new_shape.append(len(self.coords['times']))
+
+            # handle things differently if indices is defined
+            if isinstance(self.indices, tuple):
+                # TODO: improve this to be more memory efficient
+                # form all-to-all connectivity structure
+                data = np.zeros(new_shape)
+                data[:] = np.nan
+
+                row_idx, col_idx = self.indices
+                if self.is_epoched:
+                    data[:, row_idx, col_idx, ...] = self._data
+                else:
+                    data[row_idx, col_idx, ...] = self._data
+            else:
+                data = self._data.reshape(new_shape)
+
+        if squeeze:
+            return data.squeeze()
+
+        return data
+
+    def rename_nodes(self, mapping):
+        """Rename nodes.
+
+        Parameters
+        ----------
+        mapping : dict
+            Mapping from original node names (keys) to new node names (values).
+        """
+        names = self.names
+
+        # first check and assemble clean mappings of index and name
+        if isinstance(mapping, dict):
+            orig_names = sorted(list(mapping.keys()))
+            missing = [orig_name not in names for orig_name in orig_names]
+            if any(missing):
+                raise ValueError(
+                    "Name(s) in mapping missing from info: "
+                    "%s" % np.array(orig_names)[np.array(missing)])
+            new_names = [(names.index(name), new_name)
+                         for name, new_name in mapping.items()]
+        elif callable(mapping):
+            new_names = [(ci, mapping(name))
+                         for ci, name in enumerate(names)]
+        else:
+            raise ValueError('mapping must be callable or dict, not %s'
+                             % (type(mapping),))
+
+        # check we got all strings out of the mapping
+        for new_name in new_names:
+            _validate_type(new_name[1], 'str', 'New name mappings')
+
+        # do the remapping locally
+        for c_ind, new_name in new_names:
+            names[c_ind] = new_name
+
+        # check that all the channel names are unique
+        if len(names) != len(np.unique(names)):
+            raise ValueError(
+                'New channel names are not unique, renaming failed')
+
+        # rename the new names
+        self._obj.attrs['node_names'] = names
+
+    def plot_circle(self, **kwargs):
```

Comment:
There are no docs for this function. You probably want `copy_function_doc_to_method_doc` or whatever it's called from MNE